content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -210,6 +210,28 @@ When you introspect a database with the `db pull` command with the `prisma` rela
 
 When you apply changes to your Prisma schema with Prisma Migrate or `db push` with the `prisma` relation mode enabled, Prisma will not use foreign keys in your database.
 
+### Indexes
+
+In relational databases that use foreign key constraints, the database usually also implicitly creates an index for the foreign key columns. For example, [MySQL will create an index on all foreign key columns](https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html). This is to allow foreign key checks to run fast and not require a table scan.
+
+The `prisma` relation mode does not use foreign keys, so no indexes are created when you use Prisma Migrate or `db push` to apply changes to your database. You instead need to manually add an index on your relation fields with the [`@@index`](/reference/api-reference/prisma-schema-reference#index) attribute (or the [`@unique`](/reference/api-reference/prisma-schema-reference#unique), [`@@unique`](/reference/api-reference/prisma-schema-reference#unique-1) or [`@@id`](/reference/api-reference/prisma-schema-reference#id-1) attributes, if applicable).
+
+If you do not add the index manually, queries may require full table scans. This can be slow, and also expensive on database providers that bill per accessed row. To help avoid this, Prisma warns you when your schema contains fields that are used in a `@relation` that does not have an index defined:
+
+```
+TODO Good, clean example for a warning
+```
+
+In the example above you could do so by adding the following index to your model:
+
+```prisma
+@@index([foo, bar, baz])
+```
+
+If you are using the [Prisma Visual Studio Code extension](TODO link extension) (or our [language server in another editor](TODO link editor setup)) the warning will be augmented with a Quick Fix that will add the required index for you:
```

Comment:
Do we definitely want this part? Currently we don't normally document VS Code extension messages and fixes, and it could lead to a lot of screenshots to maintain if we do. I'm ok with including it if it's important in this specific case though.

Reply:
This would be important to document, it's the first time we have a recommendation warning along with a quick fix. And this one can cost you a lot of money on PlanetScale 😄

Reply:
Thanks, just wanted some context!

Reply:
Yes, this is a special feature of which we only have 1 or two - and it is _the_ easiest way to do this for a schema which might have a few hundred of these warnings. So we should do that (with the same example as above).